include/dynd/kernels/arithmetic.hpp

```diff
@@ -8,35 +8,94 @@
 namespace dynd {
 namespace nd {
 
+namespace detail{
+  template<typename>
+  struct sfinae_true : std::true_type{};
+}
```

Comment:
Things like sfinae_true should go into config.hpp as they are more general. But, we should be really careful about adding things like that.

sfinae_true seems like something you need, but there is a true_t on config.hpp which does the same thing. Probably best to use that.